### PR TITLE
Readonly queries

### PR DESCRIPTION
**Pull request by jayrbolton**
Make all query requests read-only.

We probably want the ability to do updates in views in the future. But let's do that when we need it.

- [x] I updated the README.md docs to reflect this change.
- [x] This is either not a breaking API change, or I incremented the API version.

### CODE REVIEW

File: src/relation_engine_server/utils/arango_client.py [hunk 1]
```diff
@@ -108,6 +110,40 @@ def import_from_file(file_path, query):
...
+        auth=(config['db_user'], config['db_pass'])
+    )
+    if resp.status_code == 200:
+        return
+    # Create the user
+    resp = requests.post(
+        config['db_url'] + '/_api/user',
+        data=json.dumps({'user': user, 'passwd': config['db_readonly_user']}),
+        auth=(config['db_user'], config['db_pass'])
+    )
```
Comment (JamesJeffryes) [inline src/relation_engine_server/utils/arango_client.py]: I like that this makes deployment easy as the user does not need to already exist. On the other hand, I assume this requires root access which requires the service to have the root privileges. I had thought we might want more limited privs for the api user like write access to the particular environment databases

Comment (jayrbolton) [reply src/relation_engine_server/utils/arango_client.py]: We could do that. db_user and db_pass can be set to a non-root user with 'rw' access to the specific database. That user will be able to create and grant 'ro' access for another user on the same database. Does that sound good? Doesn't require code changes

Comment (JamesJeffryes) [reply src/relation_engine_server/utils/arango_client.py]: That's good. From the docs I didn't find it clear that you can create a new user w/o admin on _system.

**Comment by jayrbolton** [comment]
I haven't yet figured out why travis is failing here...

**Comment by jayrbolton** [comment]
I finally found the travis problem: https://travis-ci.org/kbase/relation_engine_api/builds/506044760

I'm not up to date with `develop` and still working off `master`. Old habits die hard hard. Should be fixed soon now.

**Comment by jayrbolton** [comment]
It doesn't work in cluster mode.. interesting...

**Comment by jayrbolton** [comment]
I can't get readonly access working in cluster mode. The write seems to happen no matter what, in all these cases:
- create through the init_readonly_user with 'ro' access on the database and also on the "*" collection
- create manually through the dashboard and grant explicit readonly access on all collections
- leave out the basic auth header altogether in the run_query request. It still writes. Wut.

A little confusing/disconcerting to say the least. I'm going to post somewhere in arango help.

**Comment by jayrbolton** [comment]
There must be some docs somewhere saying something about special considerations with arangodb user permissions and cluster mode. But I can't find it...

**Comment by JamesJeffryes** [comment]
This is super weird. I'm glad we are running into this now rather than
later though.

**Comment by jayrbolton** [comment]
This should be fixed now by using a jwt secret when setting up the cluster.

I'm not sure why this would disable/enable user and collection level auth. I'm talking to the arango folks about it on slack.

**Pull request merged**